core/src/main/java/com/microsoft/applicationinsights/internal/heartbeat/MiscUtils.java

```diff
@@ -0,0 +1,69 @@
+package com.microsoft.applicationinsights.internal.heartbeat;
+
+import com.microsoft.applicationinsights.internal.logger.InternalLogger;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.lang3.exception.ExceptionUtils;
+
+/**
+ * This class contains some misc. functions used in heartbeat module.
+ *
+ * @author Dhaval Doshi
+ * @since 03-30-2018
+ */
+public class MiscUtils {
+
+  /**
+   * Returns a list which contains Items present in list2 but not in the set.
+   * @param set
+   * @param list2
+   * @return
+   */
+  public static Set<String> except(Set<String> set, List<String> list2) {
```

Comment:
This is `L - S`. I would reverse the parameter order.

Also, it may be easier to do this: `return new HashSet<String>(list).removeAll(set)`

Reply:
Fixed